### PR TITLE
chore: Fixes and Readme for python<>go interface

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: go/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    entitiesProto,
+    requestDataProto,
+    fullFeatureNames)
+
+entitiesRecord.Release()
+requestDataRecords.Release()
+
+```
+
+Additionally, we need to pass in a pair of pointers into which are set in `GetOnlineFeatures()` so that the resultant feature values can be passed back to Python (via the C layer) using zero-copy semantics.
```
Comment (kevjumba) [inline go/README.md:66]: maybe rephrase to, "Additionally, we need to pass in a pair of pointers to `GetOnlineFeatures()` that are populated by Go and the resultant populated feature values pointers(in the C layer) are passed back to Python via zero-copy semantics.

**Review by kevjumba — COMMENTED** [review]
@achals I think in `online_features_service.py` right after `resp = record_batch_to_online_response(record_batch)` we should also explicilty release the returned recordbatch. I'm not quite sure if returning from getOnlineFeatures actually decrements the reference counter but better safe than sorry?

**Comment by achals** [comment]
> @achals I think in `online_features_service.py` right after `resp = record_batch_to_online_response(record_batch)` we should also explicilty release the returned recordbatch. I'm not quite sure if returning from getOnlineFeatures actually decrements the reference counter but better safe than sorry?

I think we can explicitly `del` it

File: go/internal/feast/featurestore.go [hunk 2]
```diff
@@ -113,7 +113,7 @@ func (fs *FeatureStore) GetOnlineFeatures(
 	}
 
 	result := make([]*onlineserving.FeatureVector, 0)
-	arrowMemory := memory.NewGoAllocator()
+	arrowMemory := memory.NewCgoArrowAllocator()
```
Comment (achals) [inline go/internal/feast/featurestore.go:116]: We should fix this these lint issues

File: go/internal/feast/server/logging/memorybuffer_test.go [hunk 3]
```diff
@@ -159,7 +159,7 @@ func TestSerializeToArrowTable(t *testing.T) {
 	expectedRecord := builder.NewRecord()
 	assert.Nil(t, err)
 	for colIdx := 0; colIdx < int(record.NumCols()); colIdx++ {
-		assert.Equal(t, expectedRecord.Column(colIdx), record.Column(colIdx), "Columns with idx %d are not equal", colIdx)
+		assert.Equal(t, true, array.Equal(expectedRecord.Column(colIdx), record.Column(colIdx)), "Columns with idx %d are not equal", colIdx)
```
Comment (achals) [inline go/internal/feast/server/logging/memorybuffer_test.go:162]: https://pkg.go.dev/github.com/stretchr/testify/assert#True instead of `assert.Equal`

Comment (felixwang9817) [reply go/internal/feast/server/logging/memorybuffer_test.go]: fixed

File: go/internal/feast/featurestore.go (hunk 2 above)
Comment (felixwang9817) [reply go/internal/feast/featurestore.go:116]: I think #2940 should resolve this?

**Review by kevjumba — APPROVED** [review]
/lgtm
